src/navigation_plugin.cpp

```diff
@@ -249,70 +234,48 @@ bool NavigationPlugin::srvGetGoalConstraint(const ed_navigation::GetGoalConstrai
         }
 
         std::stringstream entity_constraint;
+        bool volume_found = false;
 
-        const tue::config::DataConstPointer& data = e->data();
-        tue::config::Reader r(data);
-        if (!data.empty() && r.readArray("areas"))
+        if(req.area_names[i] == "near")
         {
-            bool found_area = false;
-            while(r.nextArrayItem())
+            if (!e->shape())
             {
-                std::string name;
-                if (!r.value("name", name) || name != req.area_names[i])
-                    continue;
-
-                found_area = true;
-
-                if (!r.readArray("shape", tue::config::OPTIONAL))
-                {
-                    std::vector<geo::Vector3> chull; // In MAP frame
-
-                    // If no shape specified, get the convex hull of the object
-                    for (std::vector<geo::Vec2f>::const_iterator it = e->convexHull().points.begin(); it != e->convexHull().points.end(); ++it)
-                        chull.push_back(geo::Vector3(it->x + e->pose().t.x, it->y + e->pose().t.y, 0.0));
-
-                    // Default if not specified
-                    if (!r.value("offset", offset, tue::config::OPTIONAL))
-                        offset = default_offset_;
-
-                    //Th chull is already in map frame, no need to transform it from the local frame to the global /map frame
-                    constructConstraint(chull, entity_constraint, offset);
-                }
-                else
-                {
-                    std::string shape_constraint = constructShapeConstraint(r, e->pose());
-                    if (!shape_constraint.empty())
-                        entity_constraint << "(" << shape_constraint << ")";
-                    r.endArray();
-                }
+                res.error_msg = "Navigating to area 'near' of entity '" + req.entity_ids[i] + "' isn't possible, because it doesn't have a shape.";
+                continue;
             }
+            volume_found = true;
+            std::vector<geo::Vec2f> points;
 
-            if (!found_area)
-                res.error_msg = "Entity '" + e->id().str() + "': area '" + req.area_names[i] + "' does not exist";
+            for (std::vector<geo::Vec2f>::const_iterator it = e->convexHull().points.begin(); it != e->convexHull().points.end(); ++it)
+                points.push_back(geo::Vec2f(it->x + e->pose().t.x, it->y + e->pose().t.y));
+            ed::ConvexHull chull; // In MAP frame
+            ed::convex_hull::createAbsolute(points, 0., 0.1, chull);
```

Comment:
The for-loop + hull construction looks like a generic thing that happens more often. Refactor to a function of its own,

Reply:
All convexhull stuff in ed is a pretty big mess. If it should be somewhere, it should be there. But putting it their right now is useless, because of the mess. So keeping it this way for now.

Reply:
We keep a record of such "stuff to clean up" somewhere other than your head?